src/endpoints/usernames/username.service.ts

```diff
@@ -1,16 +1,36 @@
-import { AddressUtils, BinaryUtils, Constants, CachingService } from "@elrondnetwork/erdnest";
-import { Injectable } from "@nestjs/common";
+import { Constants, CachingService, ApiService, OriginLogger, BinaryUtils, AddressUtils } from "@elrondnetwork/erdnest";
+import { HttpStatus, Injectable } from "@nestjs/common";
+import { ApiConfigService } from "src/common/api-config/api.config.service";
 import { VmQueryService } from "../vm.query/vm.query.service";
 import { UsernameUtils } from "./username.utils";
 
 @Injectable()
 export class UsernameService {
+  private readonly logger = new OriginLogger(UsernameService.name);
+
   constructor(
-    private readonly vmQueryService: VmQueryService,
     private readonly cachingService: CachingService,
+    private readonly apiService: ApiService,
+    private readonly apiConfigService: ApiConfigService,
+    private readonly vmQueryService: VmQueryService
   ) { }
 
-  private async getUsernameAddress(username: string): Promise<string | null> {
+  async getUsernameForAddressRaw(address: string): Promise<string | null> {
+    try {
+      // eslint-disable-next-line require-await
+      const result = await this.apiService.get(`${this.apiConfigService.getMaiarIdUrl()}/users/api/v1/users/${address}`, undefined, async error => error?.response?.status === HttpStatus.FORBIDDEN);
```

Comment:
is the eslinter-disable required ?

Reply:
yes, because the callback is a Promise and we don't execute any await, it gives a warning
but in this particular situation, we know what we're doing 💪